core/vm/logger.go

```diff
@@ -65,36 +58,34 @@ type StructLog struct {
 	Err     error
 }
 
+type VmLogger interface {
+	CaptureState(env Environment, pc uint64, op OpCode, gas, cost *big.Int, memory *Memory, stack *stack, contract *Contract, depth int, err error);
+}
```

Comment:
In general, the 'generic' name Logger would be preferable for the interface because it is more general. 

Another option would be to call it `Tracer` (with method `Trace` or `TraceInstruction`).

Please document the interface in the form

```go
// Tracer is used to collect per-instruction information. The TraceMethod is
// called under such and such conditions and receives...
```